md2conf/converter.py

```diff
@@ -115,20 +125,64 @@ def _change_ext(path: str, target_ext: str) -> str:
 class ConfluenceStorageFormatConverter(NodeVisitor):
     "Transforms a plain HTML tree into the Confluence storage format."
 
+    path: str
     base_path: str
     links: List[str]
     images: List[str]
+    page_metadata: Dict[str, ConfluencePageMetadata]
 
-    def __init__(self, base_path: str) -> None:
+    def __init__(self, path: str, page_metadata: Dict[str, ConfluencePageMetadata] = dict()) -> None:
         super().__init__()
-        self.base_path = base_path
+        self.path = path
+        self.base_path = os.path.dirname(path)
         self.links = []
         self.images = []
+        self.page_metadata = page_metadata
 
     def _transform_link(self, anchor: ET.Element) -> ET.Element:
         url = anchor.attrib["href"]
         if not is_absolute_url(url):
+            LOGGER.debug(f"not is_absolute_url = {url} from = {self.path}")
             self.links.append(url)
+            # Convert the relative href to absolute based on relative url the base path value then
+            # look up the absolute path in the page metadata dictionary to discover
+            # the relative path within confluence that should be used
+            abs_path = os.path.abspath(
+                os.path.join(os.path.abspath(self.base_path), url)
+            )
```

Comment:
Like in a previous case, the outer call to `os.path.abspath` doesn't seem to be necessary. (Unless, of course, we want to ensure that `url` doesn't walk us out of the directory root, e.g. with `../../../path/to/dir`.)

Reply:
Without the outer abspath you can get paths like
`/Users/name/site/./wiki_archtypes/playbooks.md`